common/src/storage_manager.rs

```diff
@@ -2,14 +2,14 @@ use std::{fs, io, path::PathBuf};
 
 use uuid::Uuid;
 
-pub trait StorageManager: Clone + Sync + Send {
+pub trait StorageManager: Sync + Send {
     /// Path for a specific service build files
-    fn service_build_path<S: AsRef<str>>(&self, service_name: S) -> Result<PathBuf, io::Error>;
+    fn service_build_path(&self, service_name: String) -> Result<PathBuf, io::Error>;
 
     /// Path to folder for storing deployment files
-    fn deployment_storage_path<S: AsRef<str>>(
+    fn deployment_storage_path(
         &self,
-        service_name: S,
```

Comment:
I assume this gave some issues? Maybe we rather want to use `&str`?

Reply:
Yes, I believe having generics in the methods is not object safe, but we can probably use &str, let me check!

Reply:
Yeah, generic types in methods are not object safe: https://doc.rust-lang.org/reference/items/traits.html#object-safety